src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -24,6 +28,7 @@ internal sealed class EndpointRoutingMiddleware
             MatcherFactory matcherFactory,
             ILogger<EndpointRoutingMiddleware> logger,
             IEndpointRouteBuilder endpointRouteBuilder,
+            DiagnosticListener diagnosticListener,
```

Comment:
Add null check

Reply:
For our tests?

Reply:
for throughness, we null check the other parameters.